src/main/java/com/org/jenkins/custom/jenkins/distribution/service/PackagerController.java

```diff
@@ -34,4 +45,32 @@
             return new ResponseEntity(HttpStatus.NOT_FOUND);
         }
     }
+
+    /*
+     * Usage : POST: package/downloadPackageConfiguration
+     * Returns: Packager-config.yml
+     */
+    @PostMapping (path = "/downloadPackageConfiguration")
+    public ResponseEntity<?> downloadPackageConfig(@RequestBody String postPayload) {
+        LOGGER.info("Request Received for downloading configuration with params" + postPayload);
+        File packagerConfigFile = null;
+        try {
+            packagerConfigFile = new File("packagerConfig.yml");
+            // Write to file
+            FileWriter fileWriter = new FileWriter(packagerConfigFile);
+            fileWriter.write(postPayload);
+            fileWriter.flush();
+            fileWriter.close();
```

Comment:
I do not understand why writing to a file is done here. Can you explain please? Is it for debug purposes?

Reply:
Yeah because the method `returnResource` needs to calculate the length of the file    `.contentLength(file.length())`
Therefore we need to write to a file.

Reply:
Ah, I missed that. That is only visible if we also look at `Util.returnResource(...)`. A simplification could be to pass the length to that method.

Reply:
How about passing the value `payload.getBytes().length` ?

Reply:
It's not necessary to write data into a file. Because of it's very slow.

Reply:
Fixed it :)